File: 项目文件/frontend/src/types/project-change.ts
/** 项目修改记录类型定义 */

export interface ProjectChange {
  id: string;
  project_id: string;
  number: string;
  title: string;
  date: string;
  category_major: string;
  category_minor: string | null;
  category_detail: string | null;
  content: string | null;
  status: string;
  created_at: string;
  updated_at: string;
}

export interface ProjectChangeCreate {
  project_id: string;
  number: string;
  title: string;
  date: string;
  category_major: string;
  category_minor?: string;
  category_detail?: string;
  content?: string;
  status?: string;
}

export interface ProjectChangeUpdate {
  number?: string;
  title?: string;
  date?: string;
  category_major?: string;
  category_minor?: string;
  category_detail?: string;
  content?: string;
  status?: string;
}

export interface ProjectChangeListResponse {
  items: ProjectChange[];
  total: number;
}

File: 项目文件/backend/app/schemas/project_change.py
"""项目变更 Pydantic 模型"""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class ProjectChangeCreate(BaseModel):
    """创建项目变更请求"""

    project_id: uuid.UUID
    number: str
    title: str
    date: datetime
    category_major: str
    category_minor: str | None = None
    category_detail: str | None = None
    content: str | None = None
    status: str = "pending"


class ProjectChangeUpdate(BaseModel):
    """更新项目变更请求"""

    number: str | None = None
    title: str | None = None
    date: datetime | None = None
    category_major: str | None = None
    category_minor: str | None = None
    category_detail: str | None = None
    content: str | None = None
    status: str | None = None


class ProjectChangeResponse(BaseModel):
    """项目变更响应"""

    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    number: str
    title: str
    date: datetime
    category_major: str
    category_minor: str | None = None
    category_detail: str | None = None
    content: str | None = None
    status: str
    created_at: datetime
    updated_at: datetime


class ProjectChangeListResponse(BaseModel):
    """项目变更列表响应"""

    items: list[ProjectChangeResponse]
    total: int
